libs/damap/src/lib/widgets/internal-storage-translation-table/internal-storage-translation-table.component.ts

```diff
@@ -0,0 +1,153 @@
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.internalStorageTranslations) {
+      this.dataSource.data = this.internalStorageTranslations;
+    }
+  }
+
+  ngAfterViewInit() {
+    this.dataSource.paginator = this.paginator;
+    this.dataSource.sort = this.sort;
+
+    this.dataSource.filterPredicate = (
+      data: InternalStorageTranslation,
+      filter: string,
+    ) => data.title?.toLowerCase().includes(filter.trim().toLowerCase());
+  }
+
+  deleteStorageTranslation(id: number) {
+    this.dialog
+      .open(DeleteStorageTranslationWarningDialogComponent)
+      .afterClosed()
+      .subscribe({
+        next: response => {
+          if (response) {
+            const translation = this.internalStorageTranslations.find(
+              t => t.id === id,
+            );
+            this.backendService
+              .deleteInternalStorageTranslation(translation.storageId, id)
+              .subscribe(
+                () => {
+                  this.internalStorageTranslations =
+                    this.internalStorageTranslations.filter(t => t.id !== id);
+                  this.dataSource.data = this.internalStorageTranslations;
+                },
+                error => {
+                  // Check if HTTP code 400, if yes, last translation cannot be deleted
+                  if (error.status === 400) {
+                    this.feedbackService.error(
+                      this.translateService.instant(
+                        'http.error.storageErrors.lastTranslation',
+                      ),
+                    );
+                    return;
+                  } else {
+                    this.feedbackService.error(error.message);
+                  }
+                },
+              );
+          }
+        },
+      });
+  }
+
+  editStorageTranslation(id: number) {
+    const translation = this.internalStorageTranslations.find(t => t.id === id);
+
+    const dialogRef = this.dialog.open(
+      InternalStorageTranslationDialogComponent,
+      {
+        width: '75%',
+        maxWidth: '800px',
+        data: {
+          translation: { ...translation },
+          mode: 'edit',
+          storageId: this.selectedInternalStorageId,
+        },
+      },
+    );
+
+    dialogRef.afterClosed().subscribe(translation => {
+      if (translation) {
+        this.backendService
+          .updateInternalStorageTranslation(translation)
+          .subscribe(
+            () => {
+              this.internalStorageTranslations =
+                this.internalStorageTranslations.map(t =>
+                  t.id === translation.id ? translation : t,
+                );
+              this.dataSource.data = this.internalStorageTranslations;
+              this.feedbackService.success(
+                'http.success.storage.translations.edit',
+              );
+            },
+            error => {
+              if (error.error.message) {
+                this.feedbackService.error(error.error.message);
```

Comment:
can error be null/undefined and an error is raised for accessing `message` of it?
```suggestion
              if (error.error?.message) {
                this.feedbackService.error(error.error.message);
```